recipes/openjpeg/all/conanfile.py

```diff
@@ -0,0 +1,111 @@
+from conans import ConanFile, CMake, tools
+import os
+import glob
+
+
+class OpenjpegConan(ConanFile):
+    name = "openjpeg"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "OpenJPEG is an open-source JPEG 2000 codec written in C language."
+    topics = ("conan", "jpeg2000", "jp2", "openjpeg", "image", "multimedia", "format", "graphics")
+    options = {"shared": [True, False], "build_codec": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'build_codec': True, 'fPIC': True}
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+    homepage = "https://github.com/uclouvain/openjpeg"
+    license = "BSD 2-Clause"
+    exports_sources = ["CMakeLists.txt"]
+
+    _source_subfolder = "source_subfolder"
+
+    requires = (
+        "zlib/1.2.11",
+        "lcms/2.9",
+        "libpng/1.6.37",
+        "libtiff/4.0.9"
+    )
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions['BUILD_SHARED_LIBS'] = self.options.shared
+        cmake.definitions['BUILD_STATIC_LIBS'] = not self.options.shared
+        cmake.definitions['BUILD_PKGCONFIG_FILES'] = False
+        cmake.definitions['CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP'] = True
+
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        # ensure that bundled cmake files are not used
+        os.unlink(os.path.join(self._source_subfolder, 'cmake', 'FindLCMS.cmake'))
+        os.unlink(os.path.join(self._source_subfolder, 'cmake', 'FindLCMS2.cmake'))
+
+        # fix installing libs when only shared or static library built
+        tools.replace_in_file(os.path.join(self._source_subfolder, 'src', 'lib', 'openjp2', 'CMakeLists.txt'),
+                              'add_library(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})',
+                              'add_library(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})\n'
+                              'set(INSTALL_LIBS ${OPENJPEG_LIBRARY_NAME})')
+
+        # cmake tries to find LCMS2 library with LCMS2 API that is packaged as lcms library
+        # it should not be changed in conan-lcms because both lcms and lcms2 names are widespread
+        tools.replace_in_file(os.path.join(self._source_subfolder, 'thirdparty', 'CMakeLists.txt'),
+                              'find_package(LCMS2)',
+                              'find_package(lcms)')
+        tools.replace_in_file(os.path.join(self._source_subfolder, 'thirdparty', 'CMakeLists.txt'),
```

Comment:
No need to do it now, maybe this can be solved upstream with the `cpp_info.name` feature

Reply:
Unfortunately, it is not exactly the same case. LCMS Conan package has a cpp_info.name = 'lcms' as default.
But openjpeg tries to find one of LCMS or LCMS2 libraries. They have different APIs, lcms 1.x is obsolete now. So it is required to replace LCMS2 to lcms in openjpeg recipe. Changing lcms cpp_info.name does not help in this case because openjpeg will think it is a version 1.